### PR TITLE
Add Author and Game Classes

### PR DESCRIPTION
**Pull request by MarwanDev**
### In this milestone I did the following:

- Create a Game class in a separate .rb file.
- Create an Author class with an association to the Item class (in a separate .rb file).
- All Game class properties visible in the diagram should be defined and set up in the constructor method.
- All Author class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Author class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Game class
        - should override the method from the parent class
        - should return true if parent's method returns true AND if last_played_at is older than 2 years
        - otherwise, it should return false

### CODE REVIEW

**Comment by MarwanDev** [comment]
Hi, @ehmaddd Could @stephenkati  one of you review my PR?

File: classes/author.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Author
+  attr_accessor :name, :items
```
Comment (ehmaddd) [inline classes/author.rb:2]: First name and Last names should be separately stored in two individual variables

File: classes/author.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  attr_reader :id
+
+  def initialize(name)
+    @id = rand(1...1000)
+    @name = name
```
Comment (ehmaddd) [inline classes/author.rb:7]: First name and Last names should be separately stored in two individual variables

**Comment by MarwanDev** [comment]
@ehmaddd Changes have been applied. Feel free to review whenever you have time

File: spec/game_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    end
+  end
+  describe '#multiplayer' do
+  it 'returns the correct publish_date' do
+    @game.last_played_at.should eql "2023/07/04"
+    end
+  end
+  describe '#publish_date' do
+  it 'returns the correct publish_date' do
+    @game.publish_date.should eql '2018/06/15'
```
Comment (stephenkati) [inline spec/game_spec.rb:27]: use of `should` is **deprecated**.
I suggest you use `expect` instead

For example:
sh
```
expect(@game).to be_an_instance_of(Game)
expect(@game.multiplayer).to be true
expect(@game.last_played).to be eq('your date')
expect(@game.publish_date).to be eq('your date)
```

**Review by stephenkati — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUESTED :recycle: 
Hi @MarwanDev,

Good job so far. Just a few changes remaining.

## Highlights

- No Linter errors 👍 
- Clean Code 😄

**Review by stephenkati — APPROVED** [review]
# STATUS: APPROVED 🟢 
Hi @MarwanDev , Congratulations, Pull Request is approved and ready to be merged
#
## Highlights

- No Linter errors:heavy_check_mark:  
- Clean Code  :heavy_check_mark:

**Pull request merged**